src/components/vm/consoles/vnc.jsx

```diff
@@ -73,12 +73,16 @@ class Vnc extends React.Component {
 
         cockpit.transport.wait(() => {
             const prefix = (new URL(cockpit.transport.uri("channel/" + cockpit.transport.csrf_token))).pathname;
-            const query = JSON.stringify({
+            const data = {
                 payload: "stream",
                 binary: "raw",
                 address: consoleDetail.address,
                 port: parseInt(consoleDetail.tlsPort || consoleDetail.port, 10),
-            });
+            };
+            if (cockpit.transport.host !== "localhost") {
+                data.host = cockpit.transport.host;
+            }
```

Comment:
No need to make this conditional, please always set it. Plus, please add a link to the jira or github issue.